src/preamble.js

```diff
@@ -2223,6 +2223,7 @@ function integrateWasmJS(Module) {
       exports = instance.exports;
       if (exports.memory) mergeMemory(exports.memory);
       Module['asm'] = exports;
+      asm = exports;
```

Comment:
How about changing that code in parseTools from looking at `asm` to looking at `Module['asm']`? That should be the proper way to look, allowing for updates.